x/ecocredit/migrations/v3/core.go

```diff
@@ -41,6 +44,30 @@ func MigrateState(sdkCtx sdk.Context, storeKey storetypes.StoreKey,
 	}
 	creditTypeSeqTable := creditTypeSeqTableBuilder.Build()
 
+	// migrate credit types from params to ORM table
+	bz := subspace.GetRaw(sdkCtx, KeyCreditTypes)
+	if bz == nil {
+		return fmt.Errorf("credit types empty")
+	}
+
+	var creditTypes []CreditType
+	if err := json.Unmarshal(bz, &creditTypes); err != nil {
+		return err
+	}
+
+	ctx := sdk.WrapSDKContext(sdkCtx)
+	for _, creditType := range creditTypes {
+		if err := ss.CreditTypeTable().Insert(ctx, &api.CreditType{
+			Abbreviation: creditType.Abbreviation,
+			Name:         creditType.Name,
+			Unit:         creditType.Unit,
+			Precision:    creditType.Precision,
+		}); err != nil {
+			return err
+		}
+	}
+	// TODO: Set AskDenom parameter https://github.com/regen-network/regen-ledger/issues/980
```

Comment:
I don't think we need this todo here. The `AskDenom` parameter is a new parameter being introduced as part of the `marketplace` submodule in v4.0. Adding this new parameter should happen in the upgrade handler.

```suggestion
```

Reply:
just a heads up this will likely be an ORM operation once my gov handler PR is finished \

Reply:
> just a heads up this will likely be an ORM operation once my gov handler PR is finished

How so? These are the core migrations and marketplace is a new submodule.

Reply:
i'd imagine we want at least `uregen` in the AskDenom table when the upgrade is completed

Reply:
These are the state migrations for ecocredit core. I don't believe we need to do add anything related to ask denom here. Also, we are planning to start with allowed denoms empty on mainnet and to submit a followup proposal. We feel this is the appropriate action and the community should vote on adding REGEN to allowed denoms.

Reply:
ah, was unaware of the last part. thanks for the info 👍🏻